### ISSUE TITLE
economics - implementing narrow-table class

### ISSUE THREAD

**Issue by lwtchu** [openstax/cnx-recipes#3503]
![image](https://user-images.githubusercontent.com/11299125/107663237-c0a2a480-6c50-11eb-9136-586f016a058e.png)

I added the class `narrow-table` to the table in the screenshot.

Since Prince does not support the caption-side element yet, I think we should see if this could just set a minimum width for the table. 

Can we have it be around 1/2 of the page? I'm not sure how much v-grid that is maybe 8?

Also when we implement it'd be good to answer these questions
- can we implement this library wide? it seems like a generic enough element to utilize across the library
- what would it take for an existing book right now to use this class

**Comment by ajluk** [comment]
@lwtchu I have added the min-width: 20% to the container (div.os-narrow-table-container) not to the table so the caption will have the same width as the table and it works in html, but not in the pdf. The effect of min-width, max-width properties on tables are undefined in CSS and the container is set to `display:table. So I have tried `width: 100%` and it works then. Here is how it looks in pdf: 
![image](https://user-images.githubusercontent.com/55203598/108382897-97ae8080-7209-11eb-9b61-b6d9d059d260.png)
About implementing this solution library wide id depends on how many classes these tables will use. If only default tables will use it or also top-titled. But first to answer this question we need to spot all of the tables in the content to add the class where needed. I have also added a small fix to the recipes to create container class from this class and now all of the books can use it. What concerns me is that if this solution can be applied to all of the similar tables if it needs to be `width` set for them not `min-width`.

**Comment by ajluk** [comment]
@lwtchu I have tried another solution (using `display: flex` on the container and `max-width: 25%` on table caption). And this is how it looks then: 
![image](https://user-images.githubusercontent.com/55203598/108498950-e281d400-72ad-11eb-8934-5f78840e7382.png)
Then table remains in its own size and if the caption text will be longer it will wrap. Here the concern is that the flex property could display not properly in exercises where we use `display: table` on problem containers and in the tables which will be very long and need to break to the second page - then the flex property could move the whole table to the next page, but it neds to be checked as well if will find such table.

**Comment by lwtchu** [comment]
@ajluk are the two options right now either 1. set a min width or 2. have the caption flex?

**Comment by ajluk** [comment]
@lwtchu Yes, but **option 1** is using `width: 20%` not `min-width` and `display: table` on the table container (`os-table`) and in **option 2** the caption is using `max-width: 25%` and the table container `display: flex`.

**Comment by lwtchu** [comment]
can we have it be option 1? if we put these on existing tables with other classes is it possible for it to just do the width 20% on top of whatever else it's doing? or is that not how ti works :joy:

**Comment by ajluk** [comment]
@lwtchu I'm affraid it can't be used on all of the tables, because the `width: 20%` will be to small to the wide tables. It can be only used on narrow tables like this one from above with specific class `narrow-table` added to tag them.

**Comment by lwtchu** [comment]
@ajluk oh let me rephrase - like if i find all the narrow tables and apply taht class, do you think it'll just set the width="20%" in addition to the unnumbered/top-titled effects?

**Comment by ajluk** [comment]
Oh, so I didn't understand what you mean by all tables :sweat_smile: . Yes it is possible, but I need to add small change to recipes to let these classes be mixed for table containers classess (f.ex. `os-top-titled-narrow-table`). I can do that within one PR to make them ready to use for every table we have. Should I also enable this option for `text-heavy` tables?

**Comment by lwtchu** [comment]
@ajluk making note here from parking lot: going to implement in the general spec. will apply to this book for now. 🤔  i need to go thru the book and add the class. 😭

**Comment by ajluk** [comment]
@lwtchu I have applied the styles to all the tables with class `narrow-table`added by you and I noticed that for one of them `width=20%` (1/5 of page) was rather to small. It looked like this: 
![image](https://user-images.githubusercontent.com/55203598/109487665-cf78bc00-7a84-11eb-9467-23662943943d.png)

The rest of the tables looked like this though:
![image](https://user-images.githubusercontent.com/55203598/109487747-ee774e00-7a84-11eb-92a3-3521663e7a61.png)
![image](https://user-images.githubusercontent.com/55203598/109487760-f33c0200-7a84-11eb-9c93-3f2a186c8824.png)
![image](https://user-images.githubusercontent.com/55203598/109487805-018a1e00-7a85-11eb-8c3c-ad6274ded031.png)
![image](https://user-images.githubusercontent.com/55203598/109487816-05b63b80-7a85-11eb-8c05-fd89ca8920fa.png)

I have increased the width of the tables to 25% and then the one which was causing problems looks like this:
![image](https://user-images.githubusercontent.com/55203598/109488066-5af24d00-7a85-11eb-929e-eebcf331bb92.png)
And the rest of them:
![image](https://user-images.githubusercontent.com/55203598/109488112-69d8ff80-7a85-11eb-83d4-d92a02033f3a.png)
![image](https://user-images.githubusercontent.com/55203598/109488133-7198a400-7a85-11eb-9593-4e7d5abcc656.png)
![image](https://user-images.githubusercontent.com/55203598/109488153-79584880-7a85-11eb-9cd4-274e131436cc.png)
![image](https://user-images.githubusercontent.com/55203598/109488210-88d79180-7a85-11eb-8131-5d90dbdef438.png)

In the second attempt I have also centered the text in the `td`s (let me know if I should keep that for theose tables).
I guess the second option (width 25%) looks better, but what do you think? I know that 25% was your first thought but the first table I found was really narrow so I thought 20% will be better that's why I have used it at first. :slightly_smiling_face:

**Comment by lwtchu** [comment]
@ajluk should we make them sit wider at 50%?

**Comment by ajluk** [comment]
@lwtchu here is how they look at 50%:
![image](https://user-images.githubusercontent.com/55203598/109528601-25fdee80-7ab5-11eb-95d6-7230633b7ea3.png)
![image](https://user-images.githubusercontent.com/55203598/109528623-2dbd9300-7ab5-11eb-9480-a864a9910018.png)
![image](https://user-images.githubusercontent.com/55203598/109528651-357d3780-7ab5-11eb-99a0-eff1a219396a.png)
![image](https://user-images.githubusercontent.com/55203598/109528687-3d3cdc00-7ab5-11eb-8c71-e9856d3f9414.png)
![image](https://user-images.githubusercontent.com/55203598/109528740-48900780-7ab5-11eb-96a1-e466fddec416.png)
Let me know if you think it is a better solution even for the most narrow one - Table A1

**Comment by lwtchu** [comment]
@wilkus27 i think so! let's go with 50%

**Comment by MartaSztuka** [comment]
##### PASSED![Screenshot from 2021-03-09 13-25-20.png](https://images.zenhubusercontent.com/5cb5ec14fec3b63424e0f03c/9ddc7f74-9f9f-419c-b15a-65ffba6c6039)![Screenshot from 2021-03-09 13-26-09.png](https://images.zenhubusercontent.com/5cb5ec14fec3b63424e0f03c/fc67755c-03cc-4438-837b-a71df59d31cb)![Screenshot from 2021-03-09 13-26-31.png](https://images.zenhubusercontent.com/5cb5ec14fec3b63424e0f03c/594e7d28-c8a4-4355-ac8b-570b6287daa4)![Screenshot from 2021-03-09 13-26-47.png](https://images.zenhubusercontent.com/5cb5ec14fec3b63424e0f03c/0fd92565-c1ae-4f80-9fee-1a6247b5e87a)![Screenshot from 2021-03-09 13-27-01.png](https://images.zenhubusercontent.com/5cb5ec14fec3b63424e0f03c/cb667029-be29-478b-aae5-86b9ea1f7aa8)
